[torch.ao.quantization] include torch.qint32 for static quant

LiamZhuuu opened the pull request:
Summary: include `torch.qint32` to `activation_is_statically_quantized` and `get_quant_type` so that fakequantize with `dtype=torch.qint32` won't be skipped

Test Plan: land the code

Differential Revision: D39907496

vkuzo reviewed (approved):
lgtm, would be great to add a test if possible

jiaxuzhu92 committed:
[torch.ao.quantization] include torch.qint32 for static quant (#85860)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/85860

include `torch.qint32` to `activation_is_statically_quantized` and `get_quant_type` so that fakequantize with `dtype=torch.qint32` won't be skipped

Test Plan: updated `test_custom_module_class`

Reviewed By: vkuzo, jerryzh168

Differential Revision: D39907496

fbshipit-source-id: 89d41e91fcde59666e147b439d35baf97a7eb3a7
